### PR TITLE
Update to helmet v4

### PR DESCRIPTION
**Pull request by mcollina**
Fixes https://github.com/fastify/help/issues/225

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -129,7 +99,7 @@ test('disabling one header does not disable the other headers', (t) => {
       'x-dns-prefetch-control': 'off',
```
Comment (EvanHahn) [inline test.js:99]: If relevant, there are some other headers that are set by default. You can see that [tested in Helmet's source code](https://github.com/helmetjs/helmet/blob/c813f916a1aef82d586e3b411b7df7c6de0046de/test/index.test.ts#L18-L36).

**Comment by mcollina** [comment]
@EvanHahn Do you think you could export the `HelmetOptions` interface? So I can simplify https://github.com/fastify/fastify-helmet/blob/master/index.d.ts.

**Comment by EvanHahn** [comment]
Does the `Parameters<typeof helmet>` type work for you? If not, I can export it and publish a new release candidate.

**Comment by mcollina** [comment]
I don't think it would work. I'm getting `Readonly<HelmetOptions> | undefined` as the first parameter of `helmet`, however It does not satisfy the other constraints for `FastifyPluginOptions`.

```
   ✖  14:43  Type Readonly<HelmetOptions> | undefined does not satisfy the constraint FastifyPluginOptions.
  Type undefined is not assignable to type FastifyPluginOptions.
```

cc @Ethan-Arrowood

**Comment by mcollina** [comment]
Can you export that type as well?

**Comment by mcollina** [comment]
Overall, I have no idea on how to reuse the types from helmet, so any help on this front would be handy.

**Comment by Ethan-Arrowood** [comment]
Does HelmetOptions extend from FastifyPluginOptions? If not it won't be able to satisfy that constraint (unless you separately copy all the types in FastifyPluginOptions). I haven't looked past just this comment so there may be more details to the solution here.

**Comment by mcollina** [comment]
As I said, I have no clue about what I'm doing :). If you could push to this branch @Ethan-Arrowood that would be amazing. The types for the options are already fully typed in helmet, so we should be able to reuse that, shouldn't we?

**Comment by Ethan-Arrowood** [comment]
I'll take a stab now

**Comment by Ethan-Arrowood** [comment]
@EvanHahn yes please export the option interface!

**Comment by EvanHahn** [comment]
Will do! I'll plan to release a new release candidate later today.

**Comment by EvanHahn** [comment]
Just published `helmet@4.0.0-rc.2`, which you should be able to install with `npm install helmet@next`.

**Comment by Ethan-Arrowood** [comment]
Okay so I've done the given type change (which I believe is correct from looking at index.js). But as expected the test file is blowing up. Im assuming the api for helmet has changed in this major? Tomorrow I can look at fixing the test file but I don't totally see the purpose of all the tests if we are strictly dependent on helmet now. The tests would be more valuable on helmet side (might be a good cross project collaboration or good-first-issue on Evan's project)

**Comment by Ethan-Arrowood** [comment]
You can see the sort of change necessary in the diff from commit https://github.com/fastify/fastify-helmet/pull/84/commits/60d9c3a1496f5d16699561d3952f3c3bade2246b

**Comment by Ethan-Arrowood** [comment]
Nothing to worry about Evan. The new helmet types are 💯

Either our previous types were a little inaccurate or some things have changed. I only know this because we have a file for testing our type definitions (powered by the tsd library). To complete this PR we need to update that test file so it correctly follows your types.

I'll be able to work on this more tomorrow

**Comment by Ethan-Arrowood** [comment]
@EvanHahn question for you about contentSecurityPolicy config option. Do the directives support functions inside the Iterable? We have a test that does that and I'm not familiar enough with the library to know if should remove that or not

File: package.json [hunk 2]
```diff
@@ -40,5 +40,10 @@
   "dependencies": {
     "fastify-plugin": "^2.0.3",
     "helmet": "^4.0.0-rc.2"
+  },
+  "tsd": {
+    "compilerOptions": {
+      "esModuleInterop": false
```
Comment (mcollina) [inline package.json]: Aren't we recommeding to have this to `true` to everybody?

File: test.js (hunk 1 above)
Comment (mcollina) [reply test.js:99]: Thanks!

File: package.json (hunk 2 above)
Comment (Ethan-Arrowood) [reply package.json]: It is now set to true, and this is only for the way the test file is written. This option can be false and the users have to import the plugin in a different way (I think with import * as whatever from whatever)

**Pull request merged**